Materials/Materials-04-Validation_Elements/Materials-04-answer-key.R

```diff
@@ -1,61 +1,133 @@
-########### setup
+library(valtools)
 
-# We have been a partially completed validation packet. Lets complete it. 
+########### Exercise: instantiate a validation framework for R base
 
-library(valtools)
-setwd(here::here("Materials/Materials-04-Validation_Elements/simple_validation"))
+###### Action
+# instantiate
+# supress opening new rstudio session
+vt_create_packet("r_pharma_validation", open = FALSE)
+
+##### Discussion questions
+# 1. what files/folders were created? what looks familiar? why?
+# 2. when would it be helpful to use open = TRUE?
+# 3. where can the folder "r_pharma_validation" live? why or why not?
+
+########### Exercise: create simple requirement file
+
+###### Action
+setwd("r_pharma_validation")
+vt_use_req("req1.md",
+           username = "useR",
+           title = "programmer",
+           open = TRUE)
+
+# Paste the following text in line 7 of req1.md
+# 1. Character variables from csv import are not coerced to factors.
+
+##### Discussion questions
+# 1. Where does vt_use_req create the requirement file? 
+#    hint: why did we need to call setwd() first?
+# 2. What are the components of the roxygen header? are these 
+#    standard roxygen2 tags? are any unique to valtools?
+
+########### Exercise: update roxygen header
+
+###### Action
+# delete lines 4-5 from req1.md
+# update the editor (line 2) to "A user"
+
+##### Discussion questions
+# 1. When does it make sense to use the risk assessment tag? 
+# 2. If risk assesment tag is present, how does {valtools} treat this info? 
+#    hint:look at cheatsheet p1
+
+########### Exercise: create a requirement report from scratch
+
+###### Action
+# 1. Create a blank Rmd file in folder "r_pharma_validation" named 
+# "Requirement_report.Rmd"
+# 2. Add yml header
+#
+# ---
+# title: "Requirement Report"
+# author: "Validation Author"
+# date: "10/18/2021"
+# output: pdf_document
+# ---
+
+# 3. add the following chunk (setup)
+#
+# ```{r setup, echo=FALSE, warning=FALSE}
+# library(valtools)
+# library(knitr)
+# opts_chunk$set(echo = FALSE, 
+#                results="asis")
+# ```
+# 4. add the following chunk (vt_file)
+#
+# ```{r requirements}
+# req_path <- file.path("validation", "requirements")
+# vt_file(file.path(req_path, "req1.md"))
+# ```
+# 5. set up the validation config file. In /validation/validation.yml, 
+# update lines 4-5 to:
+#
+# report_rmd_name: Requirement_report.Rmd
+# report_naming_format: Requirement_Report_{date}
+#
+# 6. Render the requirement report
+#
+# vt_validate_report()
```

Comment:
This concept has not been introduced yet. Lets limit the rendering to using the knit button. This also then makes it so they don't need to create the validation.yml yet.

Reply:
the yml gets generated when they set up the packet. I'll switch this to an advanced/optional ex, though